python/cli.rst

```diff
@@ -0,0 +1,68 @@
+##############################
+Python Command Line Interfaces
+##############################
+
+Command line interfaces (CLI) in Python may be implemented using the ``argparse`` module or using `Click`_.
+
+.. _Click: https://click.palletsprojects.com
+
+.. _cli-using-click:
+
+Command Line Interfaces using Click
+===================================
+
+Click is a Python package for creating command line interfaces in a composable way. It supports:
+
+* arbitrary nesting of commands
+* automatic help page generation
+* lazy loading of subcommands at runtime
+
+`Click`_ has good `documentation`_ including a `quickstart`_ guide.
+
+.. _documentation: https://click.palletsprojects.com/en/7.x/#documentation
+.. _quickstart: https://click.palletsprojects.com/en/7.x/quickstart/
+
+The ``butler`` CLI command is implemented in ``daf_butler``. It has a reusalbe framework for dynamically
+loading subcommands from plugin packages. It is documented in the ``daf_butler`` package documentation.
+
+Add a Callable CLI Command To Your Package
+==========================================
+
+To create a callable command at the top level of your package create a folder called ``bin.src``, it should
+contain two files:
+
+1. ``SConscript`` with contents:
+
+.. code-block:: py
+
+    # -*- python -*-
+    from lsst.sconsUtils import scripts
+    scripts.BasicSConscript.shebang()
+
+2. A file that has the name of the CLI command the user will call. This file should contain as little
+   implementation as possible, it should import an implementation file and call a function in it. For example:
+
+    a. **Click Scripts**
+
+    A file ``bin.src/butler`` is implemented as:
+
+    .. code-block:: py
+
+        import sys
+        from lsst.daf.butler.cli.butler import main
+
+        if __name__ == '__main__':
+            sys.exit(main())
+
+    b. **Pipe Task Scripts**
```

Comment:
Why “Pipe Task”? Is the implication that Click is to be used for all scripts *except* those which are calling `parseAndRun()` on a task? If so, could you clarify that above?

In general, it's not really clear whether this section is describing requirements (“it *must* be done this way”) or suggestions (“you could do it like this if you like”).